singer-io/tap-zendesk
# Tdl 14803 check api access in discovery mode

# Description of change
TDL-14803 Check API access in Discovery mode
- In Discovery Mode, the tap makes an API call to the each stream.
- Write unittest cases for the same.
- Updated error handling. (TDL-15392)
- Replaced get call in get_incremental_export(http.py) by call_api because it doing the same as call_api. So, now error handling will work properly with backoff as earlier backoff was not available.

**Note** - Added code of PR79 in this PR because it was require. So, first PR79 need to merge and once that PR merged changes will not be reflected here.
# Manual QA steps
 - Check that tap shows error in discovery code for streams which does not have access.
 - Check that ticket_metrics, ticket_comments, ticket_audit stream skip 404 error.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
